### PR TITLE
CANS-870: Reminder to save when user uses the keyboard shortcut to navigate backwards or forwards

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
## Tests

- [ ] I used TDD to develop the feature
- [x] I have included unit tests
- [x] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/common/PageLockService.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
 import { createBrowserHistory } from 'history'
-import { basePath } from '../../util/common'
+import { basePath, getPageRoute } from '../../util/common'
 
 class PageLockService {
   // creates react history, sets page unloading confirmation
   // confirmation will be triggered ONLY IF method lock() has been called before
   constructor() {
+    const self = this
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/common/PageLockService.js]: `self` should not be necessary here. The use of arrow functions will ensure that `this` is bound lexically.

Comment (DmitriyRudenko) [reply app/javascript/Application/components/common/PageLockService.js]: done

File: spec/regression/case_worker_scenarios_spec.rb [hunk 2]
```diff
@@ -131,6 +131,85 @@
     expect(@form.header.conducted_by.value).to eq(CONDUCTED_BY)
   end
 
+  scenario 'Case worker attempts to leave assessment pressing "go back" then cancel' do
```
Comment (LeonidMarushevskyi) [inline spec/regression/case_worker_scenarios_spec.rb]: General approach we took before was to extend existing test scenarios where it is possible instead of creating many small scenarios for each corner case. Scenarios look repetitive which also will increase test execution time. Maybe you can use one assessment to test several corner cases?

File: spec/regression/case_worker_scenarios_spec.rb [hunk 3]
```diff
@@ -131,6 +131,85 @@
...
+  scenario 'Case worker clicks "go back" on completed assessment - no modal displayed' do
+    visit '/'
+    fill_form_header_6_to_21
+    expand_all_domains
+    check_redacted_checkbox_6_to_21
+    verify_caregiver_name_input_field_and_label
+    fill_out_assessment_form_and_check_domain_total
+    check_all_progress_are_fully_filled
+    click_save_and_summary_card_is_shown
+    click_complete_button_then_summary_card_shown
```
Comment (LeonidMarushevskyi) [inline spec/regression/case_worker_scenarios_spec.rb]: A lot of things we repeat here which are not a part of what we trying to test. We should incorporate this warning test into an existing scenario.

File: spec/regression/case_worker_scenarios_spec.rb (hunk 2 above)
Comment (DmitriyRudenko) [reply spec/regression/case_worker_scenarios_spec.rb]: done

File: spec/regression/case_worker_scenarios_spec.rb (hunk 3 above)
Comment (DmitriyRudenko) [reply spec/regression/case_worker_scenarios_spec.rb]: done

**Comment by DmitriyRudenko** [comment]
retest

**Comment by DmitriyRudenko** [comment]
Jenkins retest this please

**Comment by DmitriyRudenko** [comment]
Jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please
